crates/cheatnet/src/state.rs

```diff
@@ -446,6 +455,16 @@ impl TraceData {
         last_call.result = result;
         last_call.vm_trace = vm_trace;
     }
+
+    pub fn add_phantom_call(&mut self) {
```

Comment:
Not sure if using `phantom` name is a good idea here:
- `phantom` already has meaning in rust
- we use those nodes only for deploys without constructors, it is better to use a specific name than abstract

Reply:
`add_empty_call` would also work i guess

aren't there any future uses for this? btw - isn't this kind of design flaw, where we should always treat the deploy as a special case, no matter if it has the constructor or not? it'd make for more consistent code i think (and less hacky)

Reply:
>aren't there any future uses for this?

Not really

>isn't this kind of design flaw, where we should always treat the deploy as a special case, no matter if it has the constructor or not

No, we shouldn't we should only treat it as a special case in case of an empty constructor. The non-empty constructor goes through `execute_call_entry_point` so it is collected same as a regular call